test/feature/features/coordinator.feature

```diff
@@ -0,0 +1,235 @@
+    """
+    random_word
+    """
+
+  Scenario: Split/Unite key range works
+    When I run SQL on host "coordinator"
+    """
+    SPLIT KEY RANGE krid3 FROM krid1 BY 5;
+    SHOW key_ranges
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid1".*"Lower bound":"0".*"Upper bound":"5"
+    """
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid3".*"Lower bound":"5".*"Upper bound":"10"
+    """
+
+    When I run SQL on host "router-admin"
+    """
+    SHOW key_ranges
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid1".*"Lower bound":"0".*"Upper bound":"5"
+    """
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid3".*"Lower bound":"5".*"Upper bound":"10"
+    """
+
+    When I run SQL on host "coordinator"
+    """
+    UNITE KEY RANGE krid1 WITH krid3;
+    SHOW key_ranges
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid1".*"Lower bound":"0".*"Upper bound":"10"
+    """
+
+    When I run SQL on host "router-admin"
+    """
+    SHOW key_ranges
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    "Key range ID":"krid1".*"Lower bound":"0".*"Upper bound":"10"
+    """
+
+  Scenario: Split/Unite locked key range
+    When I run SQL on host "coordinator"
+    """
+    LOCK KEY RANGE krid1;
+    SPLIT KEY RANGE krid3 FROM krid1 BY 5;
+    """
+    Then SQL error on host "coordinator" should match regexp
+    """
+    context deadline exceeded
+    """
+
+    When I run SQL on host "coordinator"
+    """
+    UNITE KEY RANGE krid1 WITH krid2
+    """
+    Then SQL error on host "coordinator" should match regexp
+    """
+    context deadline exceeded
+    """
+
+  Scenario: Router is down
+    #
+    # Coordinator doesn't unregister router
+    #
+    Given host "router" is stopped
+    When I run SQL on host "coordinator"
+    """
+    SHOW routers
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    router r1-regress_router:7000
+    """
+
+    #
+    # Coordinator doesn't crash on action
+    #
+    Given I run SQL on host "coordinator"
+    """
+    ADD KEY RANGE krid3 FROM 31 TO 40 ROUTE TO sh1
+    """
+
+    Given host "router" is started
+    When I run SQL on host "router-admin"
+    """
+    SHOW key_ranges
+    """
+    Then command return code should be "0"
+    And SQL result should match regexp
+    """
+    \[\]
+    """
+
+  Scenario: QDB is down
+    Given host "qdb01" is stopped
+    When I run SQL on host "coordinator"
+    """
+    CREATE KEY RANGE krid3 FROM 31 TO 40 ROUTE to sh1;
+    """
+    Then command return code should be "1"
+    And SQL error on host "coordinator" should match regexp
+    """
+    context deadline exceeded
```

Comment:
Maybe we should explicitly say that QDB is not available?

Reply:
Yeah, I agree, because `context deadline exceeded` is the error we get because we wait for query response for too long. It happens in many different ways, for instance, when something is locked (key range) or down (qdb).
Now goroutine, processing the query, just hangs until that something is unlocked or up again. I thought maybe we should check if required object is locked or down and if so return an error with appropriate message. 
What you think? I can try to do it in the next pr.

Reply:
> maybe we should check if required object is locked or down

agree

> I can try to do it in the next pr.

lets do this